test/support/test_config.h

```diff
@@ -132,6 +132,8 @@
 #define _PSTL_CLANG_TEST_COMPLEX_ATAN_IS_CASE_BROKEN __clang__
 #define _PSTL_CLANG_TEST_COMPLEX_SIN_IS_CASE_BROKEN __clang__
 
+#define TEST_DYNAMIC_SELECTION_AVAILABLE (__INTEL_LLVM_COMPILER >= 20230000)

```

Comment:
Should we analyze ```TEST_DPCPP_BACKEND_PRESENT``` define in device selection SYCL tests?
Usually all usages of ```sycl::queue``` in tests is under ```TEST_DPCPP_BACKEND_PRESENT``` define as I seen.

Reply:
So I propose to rewrite this line like
```C++
#define TEST_DYNAMIC_SELECTION_AVAILABLE (__INTEL_LLVM_COMPILER >= 20230000 && TEST_DPCPP_BACKEND_PRESENT )
```

Reply:
If we apply ```TEST_DPCPP_BACKEND_PRESENT``` into definition of the ```TEST_DYNAMIC_SELECTION_AVAILABLE``` I believe these block is not required anymore in the ```./test/parallel_api/dynamic_selection/sycl``` folder:
```C++
    catch (const sycl::exception&)
    {
        std::cout << "SKIPPED: Unable to use sycl at all\n";
        return 0;
    }
```